doc/docs/deploy-manage/deploy/environment-variables.md

```diff
@@ -0,0 +1,166 @@
+| `WORKER_SIDECAR_IMAGE` | `"`               | The `pachd` image that is used as a worker sidecar. |
+| `WORKER_IMAGE_PULL_POLICY` | `"`           | The pull policy that defines how Docker images are pulled. The default value is `IfNotPresent`. You can set a Kubernetes image pull policy as needed. |
+| `LOG_LEVEL`          | `info`              | Verbosity of the log output. If you want to disable logging, set this variable to `0`. |
+| `IAM_ROLE`           | `"`                 | The role that defines permissions for Pachyderm in AWS. |
+| `IMAGE_PULL_SECRET`  | `"`                 | The Kubernetes secret for image pull credentials. |
+| `NO_EXPOSE_DOCKER_SOCKET` | `false`        | Controls whether you can build images using the `--build` command. |
+| `EXPOSE_OBJECT_API`  | `false`             | Controls access to internal Pachyderm API. |
+| `PACHD_MEMORY_REQUEST` | `1T`              | ??? |
+| `WORKER_USES_ROOT`   | `true`              | Controls root access in the worker container. |
+| `S3GATEWAY_PORT`     | `600`               | The S3 gateway port number. |
+
+**Storage Configuration**
+
+| Environment Variable       | Default Value     | Description |
+| -------------------------- | ----------------- | ----------- |
+| `STORAGE_MEMORY_THRESHOLD` | N/A               | Defines the storage memory threshold. |
+| `STORAGE_SHARD_THRESHOLD`  | N/A               | Defines the storage shard threshold.  |
+
+## Pipeline Worker Environment Variables
+
+Pachyderm defines many environment variables for each Pachyderm
+worker that runs your pipeline code. You can print the list
+of environment variables into your Pachyderm logs by including
+the `env` command into your pipeline specification. For example,
+if you have an `images` repository, you can configure your pipeline
+specification like this:
+
+```json
+{
+    "pipeline": {
+        "name": "env"
+    },
+    "input": {
+        "pfs": {
+            "glob": "/",
+            "repo": "images"
+        }
+    },
+    "transform": {
+        "cmd": ["sh" ],
+        "stdin": ["env"],
+        "image": "ubuntu:14.04"
+    },
+    "enable_stats": true
+}
+```
+
+Run this pipeline and upon completion you can view the log with
+variables by running the following command:
+
+```bash
+$ pachctl logs --pipeline=env
+PPS_WORKER_IP=172.17.0.7
+DASH_PORT_8081_TCP_PROTO=tcp
+PACHD_PORT_600_TCP_PORT=600
+KUBERNETES_SERVICE_PORT=443
+KUBERNETES_PORT=tcp://10.96.0.1:443
+...
+```
+
+You should see a lengthy list of variables. Many of them define
+internal networking parameters that most probably you will not
+need to use.
+
+Most users find the following environment variables
+particularly useful:
+
+| Environment Variable       | Description |
+| -------------------------- | --------------------------------------------- |
+| `PPS_WORKER_IP`            | The IP address of the worker pod. For example, `PPS_WORKER_IP=172.17.0.7`.             |
```

Comment:
I don't think we want to expose this to users. The Pachyderm worker API is what we use to mark datums completed and such.